python/cudf/cudf/core/column_accessor.py

```diff
@@ -372,8 +372,15 @@ def insert(
             new_values = self.columns[:loc] + (value,) + self.columns[loc:]
             self._data = dict(zip(new_keys, new_values))
         self._clear_cache(old_ncols, old_ncols + 1)
-        # The type(name) may no longer match the prior label_dtype
-        self.label_dtype = None
+        if cudf.get_option("mode.pandas_compatible"):
```

Comment:
We might as well always try this, even not in pandas_compatible mode

Reply:
Done